server/src/weaverbird/backends/pypika_translator/translators/base.py

```diff
@@ -238,45 +250,129 @@ def aggregate(
         step: "AggregateStep",
     ) -> StepContext:
         agg_selected: list[Field] = []
+        window_selected: list[tuple[int, Field]] = []
+        window_subquery_list: list[Tables] = []
+
+        def _build_window_subquery() -> Any:
+            min_window_index = min([c[0] for c in window_selected])
+            first_wq = Table(f'wq{min_window_index}')
+            merged_query = (
+                self.QUERY_CLS.from_(window_subquery_list[0])
+                .select(
+                    *step.on,
+                    *[
+                        getattr(first_wq, col[1].alias)
+                        for col in window_selected
+                        if col[0] == min_window_index
+                    ],
+                )
+                .as_('window_subquery')
+            )
+            if len(window_subquery_list) > 1:
+                for index, sq in enumerate(window_subquery_list[1:]):
+                    wq_temp = Table(f'wq{min_window_index + index + 1}')
+                    merged_query = (
+                        merged_query.join(sq)
+                        .on_field(*step.on)
+                        .select(
+                            *[
+                                getattr(wq_temp, col[1].alias)
+                                for col in window_selected
+                                if col[0] == min_window_index + index + 1
+                            ]
+                        )
+                        .as_('window_subquery')
+                    )
+            return merged_query
+
+        # Handle aggregation and analytics functions in distinct subqueries
+
+        for step_index, aggregation in enumerate(step.aggregations):
+            if agg_fn := self._get_aggregate_function(aggregation.agg_function):
+                for agg_index, agg_column_name in enumerate(aggregation.columns):
+                    column_field: Field = Table(prev_step_name)[agg_column_name]
+                    new_agg_col = agg_fn(column_field).as_(aggregation.new_columns[agg_index])
```

Comment:
tip: `zip` seems to be what you want here (I'm fine with `enumerate`, it's just in case you didn't know about it :slightly_smiling_face: )

```suggestion
                for agg_column_name, new_column_name in zip(aggregation.columns, aggregation.new_columns):
                    column_field: Field = Table(prev_step_name)[agg_column_name]
                    new_agg_col = agg_fn(column_field).as_(new_column_name)
```

Reply:
thx, clearer like this :+1: